docs/deployAutomationForFirewallCostOptimization.md

```diff
@@ -0,0 +1,427 @@
+
+<img src="./assets/automation/createAutomationAccount1Portal.png" alt="Create a User-Assigned Managed Identity for the Automation" />
+
+automation account name: 'aa-ngsa-asb-firewall-automation-dev' Note, slightly different from the image above.
+
+<img src="./assets/automation/createAutomationAccount2Portal.png" alt="Create a User-Assigned Managed Identity for the Automation" />
+
+Select both the system-assigned, as well as the user-assigned managed identities check boxes as well as select the 'mi-ngsa-asb-firewall-automation-dev' managed identity to associate it with the Automation Account.
+
+Click create when finished.  
+
+#### Create Automation PowerShell Runbook
+
+<img src="./assets/automation/createAutmationRunbook1Portal.png" alt="Create a User-Assigned Managed Identity for the Automation" />
+
+Navigate to the automation account created, 'aa-ngsa-asb-firewall-automation-dev' and select the 'Runbook' blade, and "Create a runbook.  
+
+<img src="./assets/automation/createAutmationRunbook2Portal.png" alt="Create a User-Assigned Managed Identity for the Automation" />
+
+Create an Azure Powershell Runbook from the Azure Portal within the Automation Resource Group or using the Azure CLI
+
+Use the portal UX upload the Powershell runbook file located in the scripts/automation directory
+
+#### Create Automation Schedules
+
+<img src="./assets/automation/createAutmationAccountSchedule1Portal.png" alt="Create a User-Assigned Managed Identity for the Automation" />
+
+Select the "Schedules" blade on the left side of the Autoamtion Account screen.
+
+<img src="./assets/automation/createRunbookSchedule1Portal.png" alt="Create a User-Assigned Managed Identity for the Automation" />
+
+Establish the schedule desired for the firewall automation.  The Schedule decides for this implementation was to turn the development firewall off at 10:00 PM EST and back on at 6:00 AM EST.  
+
+<img src="./assets/automation/editRunbookScheduleLinked1Portal.png" alt="Create a User-Assigned Managed Identity for the Automation" />
+
+Link the runbook and the schedule
+
+<img src="./assets/automation/endAutomationScheduleLinked1Portal.png" alt="Create a User-Assigned Managed Identity for the Automation" />
+
+Once linked, the defaulted values located within the Runbook will be visible.  
+
+### Azure CLI & Az PowerShell Modules
+
+#### Create the Resource Group with the Azure CLI
+
+The following commands authenticate a user to a specified tenant and then switches into the desired subscription.  
+
+Once authentication and authorization is established the Azure CLI is used to create a resource group.  
+
+   ```bash
+
+    # Log into the tenant
+    az login --tenant {tenant Id}
+
+    # Set the subscription desired    
+    az account set --subscription {subscription name}
+    
+    # pass in resource group  name and location
+    az group create -name {automation resource group name} -location {location}
```

Comment:
Replace the `{tenant Id}` style replacements with variables. That way, it is easier to copy and paste.
The top of each code block could have all the variables needed in one spot for easy replacement.

ex:
```suggestion

    ASB_TENANT_ID={tenant Id}
    ASB_SUB_NAME={subscription name}
    ASB_FW_AUTO_LOCATION={location}
    ASB_FW_AUTO_RG={automation resource group name}

    # Log into the tenant
    az login --tenant $ASB_TENANT_ID

    # Set the subscription desired    
    az account set --subscription $ASB_SUB_NAME
    
    # pass in resource group  name and location
    az group create -name $ASB_FW_AUTO_RG -location $ASB_FW_LOCATION
```

Reply:
completed this.